openssl/src/ssl/mod.rs

```diff
@@ -2796,6 +2766,109 @@ impl<S: Read + Write> Write for SslStream<S> {
     }
 }
 
+/// A partially constructed `SslStream`, useful for unusual handshakes.
+pub struct SslStreamBuilder<S> {
+    inner: SslStream<S>
+}
+
+impl<S> SslStreamBuilder<S>
+    where S: Read + Write
+{
+    /// Begin creating an `SslStream` atop `stream`
+    pub fn new(ssl: Ssl, stream: S) -> Self {
+        Self {
+            inner: SslStream::new_base(ssl, stream),
+        }
+    }
+
+    /// Perform a stateless server-side handshake
+    ///
+    /// Requires that cookie generation and verification callbacks were
+    /// set on the SSL context.
+    ///
+    /// Returns `Ok` if a complete ClientHello containing a valid cookie was
+    /// read, in which case the handshake should be continued via `accept`. If
+    /// the cookie is missing or invalid, a HelloRetryRequest containing a new
+    /// cookie will be transmitted and `Err` returned.
+    ///
+    /// This corresponds to [`SSL_stateless`]
+    ///
+    /// [`SSL_stateless`]: https://www.openssl.org/docs/manmaster/man3/SSL_stateless.html
+    #[cfg(ossl111)]
+    pub fn stateless(&mut self) -> Result<(), Error> {
+        if (unsafe { ffi::SSL_stateless(self.inner.ssl.as_ptr()) }) != 0 {
+            Ok(())
+        } else {
+            Err(self.inner.make_error(0))
```

Comment:
Given that SSL_stateless doesn't seem to have interesting return values, should this go straight to `ErrorStack::get()` instead?